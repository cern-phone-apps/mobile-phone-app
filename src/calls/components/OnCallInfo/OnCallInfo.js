import { View } from 'react-native';
import React, { useState } from 'react';
import { Text } from 'react-native-elements';
import PropTypes from 'prop-types';
import HangupButton from '../HangupButton/HangupButton';
import Timer from '../Timer/Timer';
import styles from './OnCallInfoStyles';
import DialpadButton from './onCallDialpad';

function OnCallInfo({ remote, phoneService }) {
  const [dialpad, setDialpad] = useState(0);

  return (
    <View style={styles.container}>
      <Text style={styles.ongoingCall}>ONGOING CALL WITH</Text>
      <View style={styles.remoteInfo}>
        {remote && remote.name && (
          <Text style={styles.remoteName}>{remote.name}</Text>
        )}
        <Text style={styles.remoteNumber}>{remote.phoneNumber}</Text>
        <Timer />
      </View>
      <DialpadButton
        display={dialpad}
        onPress={() => {
          setDialpad(!dialpad);
        }}
        updatePhoneNumber={e => {
          phoneService.sendDtmfCommand(e);
        }}
      />
      <View
        style={{
          position: 'absolute',
          bottom: '5%'
        }}
      >
        <HangupButton />
      </View>
    </View>
  );
}

OnCallInfo.propTypes = {
  remote: PropTypes.shape({
    name: PropTypes.string.isRequired,
    phoneNumber: PropTypes.string.isRequired
  }).isRequired,
  phoneService: PropTypes.shape({
    sendDtmfCommand: PropTypes.func,
  }).isRequired
};

export default OnCallInfo;
