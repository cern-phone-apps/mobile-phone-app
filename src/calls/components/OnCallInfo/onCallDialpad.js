import { View, StyleSheet, TouchableOpacity } from 'react-native';
import React from 'react';
import { Icon } from 'react-native-elements';
import Dialpad from '../DialpadForm/Dialpad/Dialpad';
import ColorPalette from '../../../styles/ColorPalette';

const dialpadstyles = StyleSheet.create({
  button: {
    marginTop: 10,
    marginBottom: 10
  },
  DialpadButtonContainer: {
    alignItems: 'center',
    top: '35%'
  },
  DialpadButton: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 70,
    height: 70,
    backgroundColor: ColorPalette.primary,
    borderRadius: 50
  },
  CloseDialpadButton: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 40,
    height: 40,
    backgroundColor: ColorPalette.primary,
    borderRadius: 50
  }
});

function DialpadButton({ display, onPress, updatePhoneNumber }) {
  if (display)
    return (
      <View
        style={{
          width: '100%',
          top: '30%'
        }}
      >
        <View
          style={{
            alignItems: 'center',
            width: '100%',
            marginBottom: '1%'
          }}
        >
          <TouchableOpacity
            style={dialpadstyles.CloseDialpadButton}
            testID={"TouchableOpacity1"}
            onPress={() => {
              onPress();
            }}
          >
            <Icon name="close" size={15} color="white" type="ionicons" />
          </TouchableOpacity>
        </View>
        <Dialpad
          height="60%"
          updatePhoneNumber={e => {
            updatePhoneNumber(e);
          }}
        />
      </View>
    );
  return (
    <View style={dialpadstyles.DialpadButtonContainer}>
      <TouchableOpacity
        testID={"TouchableOpacity2"}
        style={dialpadstyles.DialpadButton}
        onPress={() => {
          onPress();
        }}
      >
        <Icon name="dialpad" size={25} color="white" type="ionicons" />
      </TouchableOpacity>
    </View>
  );
}

export default DialpadButton;
