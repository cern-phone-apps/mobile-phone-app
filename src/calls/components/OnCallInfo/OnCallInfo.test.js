import React from 'react';
import { render } from '@testing-library/react-native';
import OnCallInfo from './OnCallInfo';

const phoneService = {
  sendDtmfCommand: jest.fn(() => {})
};

// THIS TEST LAUNCH A WARNING IN HIS CHILD COMPONENTS
test('Render the component', async () => {
  const { baseElement } = render(
    <OnCallInfo
      remote={{
          name: "Name",
          phoneNumber: "64446"
      }}
      phoneService={phoneService}
    />
  );
  expect(baseElement).toMatchSnapshot();
});
