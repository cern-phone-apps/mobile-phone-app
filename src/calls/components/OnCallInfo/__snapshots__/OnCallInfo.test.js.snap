// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the component 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "opacity": 0.6,
            "position": "absolute",
            "top": "5%",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        ONGOING CALL WITH
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "position": "absolute",
            "top": "10%",
            "transform": Array [
              Object {
                "translateY": 0.5,
              },
            ],
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Name
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          64446
        </Text>
        <Text>
          00:00
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "top": "35%",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#22529e",
              "borderRadius": 50,
              "height": 70,
              "justifyContent": "center",
              "width": 70,
            }
          }
          testID="TouchableOpacity2"
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 25,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "bottom": "5%",
            "position": "absolute",
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 50,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.3}
            disabled={false}
            testID="HangupForm"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f12121",
                  "borderColor": "#2089dc",
                  "borderRadius": 50,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "center",
                  "marginTop": "20%",
                  "padding": 8,
                  "width": 70,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                  underlayColor="white"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 30,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Warning: Failed prop type: The prop \`phoneService\` is marked as required in \`HangupForm\`, but its value is \`undefined\`.
    in HangupForm (created by ThemeComponent)
    in ThemeComponent (created by OnCallInfo)
    in View (created by View)
    in View (created by OnCallInfo)
    in View (created by View)
    in View (created by OnCallInfo)
    in OnCallInfo
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Warning: Failed prop type: The prop \`phoneService\` is marked as required in \`HangupForm\`, but its value is \`undefined\`.
    in HangupForm (created by ThemeComponent)
    in ThemeComponent (created by OnCallInfo)
    in View (created by View)
    in View (created by OnCallInfo)
    in View (created by View)
    in View (created by OnCallInfo)
    in OnCallInfo
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)",
                  "substitutions": Array [],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 20,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 82,
                    "methodName": "checkPropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1796,
                    "methodName": "validatePropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/fiser/developer/mobile-phone-app/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1969,
                    "methodName": "Object.createElementWithValidation [as createElement]",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 14,
                    "file": "/Users/fiser/developer/mobile-phone-app/src/calls/providers/PhoneProvider/PhoneService.js",
                    "lineNumber": 15,
                    "methodName": "ThemeComponent.render",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 48.5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Warning: Failed prop type: The prop \`phoneService\` is marked as required in \`HangupForm\`, but its value is \`undefined\`.
    in HangupForm (created by ThemeComponent)
    in ThemeComponent (created by OnCallInfo)
    in View (created by View)
    in View (created by OnCallInfo)
    in View (created by View)
    in View (created by OnCallInfo)
    in OnCallInfo
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;
